### ISSUE TITLE
[Error] make[2]: *** [vrrp_iproute.o] Error 1

### ISSUE THREAD

**Issue by Travis-z** [acassen/keepalived#932]
When I compile, I get the following error:
make
... ...
Making all in keepalived
make[1]: Entering directory `/home/xxx/keepalived-2.0.4/keepalived'
Making all in core
make[2]: Entering directory `/home/xxx/keepalived-2.0.4/keepalived/core'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/home/xxx/keepalived-2.0.4/keepalived/core'
Making all in vrrp
make[2]: Entering directory `/home/xxx/keepalived-2.0.4/keepalived/vrrp'
  CC       vrrp_iproute.o
In file included from /usr/include/linux/rtnetlink.h:5,
                 from ./../include/vrrp_iproute.h:34,
                 from vrrp_iproute.c:26:
/usr/include/linux/netlink.h:35: error: expected specifier-qualifier-list before ‘sa_family_t’
make[2]: *** [vrrp_iproute.o] Error 1
make[2]: Leaving directory `/home/xxx/keepalived-2.0.4/keepalived/vrrp'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/xxx/keepalived-2.0.4/keepalived'
make: *** [all-recursive] Error 1

centos 6.5, keepalived-2.0.4

Have you any idea to help us to fix this problems ?
Regards

**Comment by pqarmitage** [comment]
It appears to be a problem with the kernel headers, whereby <linux/rtnetlink.h>, which needs <sys/socket.h> to be included before it is included. The following patch resolves it for me on CentOS 6.5:
```
diff --git a/keepalived/include/vrrp_iproute.h b/keepalived/include/vrrp_iproute.h
index 51f8f636..123800c6 100644
--- a/keepalived/include/vrrp_iproute.h
+++ b/keepalived/include/vrrp_iproute.h
@@ -31,6 +31,7 @@
 #if HAVE_DECL_LWTUNNEL_ENCAP_MPLS
 #include <linux/mpls.h>
 #endif
+#include <sys/socket.h>
 #include <linux/rtnetlink.h>
 
 /* local includes */
```

I'll look at producing a patch for this.

**Comment by pqarmitage** [comment]
Commit 9b05533 resolves this problem.
